src/NUnitFramework/tests/Constraints/PropertyTests.cs

```diff
@@ -206,4 +208,68 @@ public void FailureMessageContainsChainedConstraintMessage()
             Assert.IsTrue(textMessageWriter.ToString().Contains("Extra"));
         }
     }
+
+    [TestFixture(TestOf = typeof(PropertyConstraint))]
+    public class PropertyHierarchyTests
```

Comment:
I like the idea of either having the tests in the same test class or starting this class name with `PropertyConstraint` and using the refactoring to move it to its own file. Maybe `TypeHierarchy` too instead of just `Hierarchy` to be extra clear. What do you think?

Reply:
Moving to separate file is probably cleaner approach